<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1311</width>
    <height>758</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Stack interpreter</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QComboBox" name="instructions_select">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>230</y>
      <width>211</width>
      <height>25</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>Choose a instruction...</string>
    </property>
   </widget>
   <widget class="QPushButton" name="exec_button">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>260</y>
      <width>211</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Execute instruction</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="stackostream">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>200</y>
      <width>401</width>
      <height>25</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="flagged_values">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>230</y>
      <width>181</width>
      <height>25</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_stack">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>170</y>
      <width>211</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Stack</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_ins_log">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>0</y>
      <width>121</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Instructions log</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="instruction_log">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>20</y>
      <width>401</width>
      <height>661</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QProgressBar" name="stack_pb">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>370</y>
      <width>401</width>
      <height>23</height>
     </rect>
    </property>
    <property name="value">
     <number>24</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_pb_stack">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>340</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Stack size occupied</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="memostream">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>20</y>
      <width>351</width>
      <height>661</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>0</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Memory</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cpp_export_button">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>520</y>
      <width>111</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Export to .cpp</string>
    </property>
   </widget>
   <widget class="QPushButton" name="asm_export_button">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>520</y>
      <width>111</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Export to .asm</string>
    </property>
   </widget>
   <widget class="QPushButton" name="options_button">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>260</y>
      <width>181</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Size options</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>490</y>
      <width>121</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Export to  a file</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clear_memory_log_button">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>690</y>
      <width>131</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear memory log</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clear_instructions_log_button">
    <property name="geometry">
     <rect>
      <x>1050</x>
      <y>690</y>
      <width>161</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear instructions log</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1311</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="mainmenu">
    <property name="title">
     <string>Menu</string>
    </property>
    <addaction name="about"/>
   </widget>
   <addaction name="mainmenu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="about">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
